modules/gapi/perf/common/gapi_video_perf_tests_inl.hpp

```diff
@@ -0,0 +1,96 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_VIDEO_PERF_TESTS_INL_HPP
+#define OPENCV_GAPI_VIDEO_PERF_TESTS_INL_HPP
+
+#include <iostream>
+
+#include "gapi_video_perf_tests.hpp"
+
+namespace opencv_test
+{
+
+  using namespace perf;
+
+//------------------------------------------------------------------------------
+
+PERF_TEST_P_(OptFlowLKPerfTest, TestPerformance)
+{
+    std::vector<cv::Point2f> outPtsOCV,    outPtsGAPI,    inPts;
+    std::vector<uchar>       outStatusOCV, outStatusGAPI;
+    std::vector<float>       outErrOCV,    outErrGAPI;
+
+    compare_forOptFlow cmpF;
+    OptFlowLKTestParams params;
+    std::tie(cmpF, params.fileNamePattern, params.channels,
+             params.pointsNum, params.winSize, params.criteria,
+             params.compileArgs) = GetParam();
+
+    OptFlowLKTestOutput outOCV  { outPtsOCV,  outStatusOCV,  outErrOCV };
+    OptFlowLKTestOutput outGAPI { outPtsGAPI, outStatusGAPI, outErrGAPI };
+
+    cv::GComputation c = runOCVnGAPIOptFlowLK(*this, inPts, params, outOCV, outGAPI);
+
+    declare.in(in_mat1, in_mat2, inPts).out(outPtsGAPI, outStatusGAPI, outErrGAPI);
+
+    TEST_CYCLE()
+    {
+        c.apply(cv::gin(in_mat1, in_mat2, inPts, std::vector<cv::Point2f>{ }),
+                cv::gout(outPtsGAPI, outStatusGAPI, outErrGAPI));
+    }
+
+    // Comparison //////////////////////////////////////////////////////////////
+    EXPECT_TRUE(static_cast<compare_vector_forOptFlow<cv::Point2f>>(cmpF)(outPtsGAPI, outPtsOCV));
+    EXPECT_TRUE(static_cast<compare_vector_forOptFlow<uchar>>(cmpF)(outStatusGAPI, outStatusOCV));
+    EXPECT_TRUE(static_cast<compare_vector_forOptFlow<float>>(cmpF)(outErrGAPI,    outErrOCV));
```

Comment:
I don't really get what happens here.

You're casting `cmpF` to three different types??

Why do you have three comparison functions here?

Reply:
GTest is composable and it doesn't need all its `EXPECT*`/`ASSERT*` macros be called from the test body.

You can call these macros in any function (e.g. your comparison one) and then call it from your test body.

EXPECT_TRUE is not very informative as it is, when arrays or images are compared.

Reply:
I refused to cast cmpF, it looked weird; the comparison function which is called different templates is implemented instead.
As for EXPECT_TRUE, we are comparing on bitexact here, so I think it's OK not to be very informative..

Reply:
So what is `cmpF` here and how do you cast it to the different kinds of functions?

Why not finally make `cmpF` of six arguments?

Reply:
This place is rewritten now. Yes, you are right, I've probably over-complicated it. In the end, after Asya's review, I sorted it out and then implemented just what you've said - the function with all 6 vectors to compare (but it calls vector2vector compare function 3 times). Now it looks much better (for me)
Is it good just not to give cmpF to test instances as a parameter?  the case is too custom and also there is no need to check different comparisons (eg Exact and Tolerant ones) at the same time